### PR TITLE
Reduce number of appicon flickering cases

### PR DESCRIPTION
**Pull request by trunkmaster**
Override NSWindow's `becomesKeyOnlyIfNeeded` and NSView `needsPanelToBecomeKey` in appicon implementation to reduce number of order front calls. The source of the calls is NSWindow's sendEvent:.
Steps to reproduce includes: double-click on appicon of inactive application, activate visible windows of inactive application (click on titlebar or window contents). 
Result: appicon flickers (redraws). It can be observed in `cairo` backend. `art` backend doesn't show appicon flickering.

### CODE REVIEW

File: Source/NSApplication.m [hunk 1]
```diff
@@ -542,6 +547,11 @@ - (BOOL) acceptsFirstMouse: (NSEvent*)theEvent
   return YES;
 }
 
+- (BOOL) needsPanelToBecomeKey
```
Comment (fredkiefer) [inline Source/NSApplication.m]: I don't see why this change is needed. NSView already should return NO for this method so we are just duplicating code from the super class.
The other part of this change looks good to me.

Comment (trunkmaster) [reply Source/NSApplication.m]: OK, let me check NSView code.

Comment (trunkmaster) [reply Source/NSApplication.m]: You are right. I've removed `needsPanelToBecomeKey` override.

Comment (trunkmaster) [reply Source/NSApplication.m]: By the way, what window manager do you use? I need to know to do additional checks before submitting PR to -gui and -back.

Comment (fredkiefer) [reply Source/NSApplication.m]: I normally use kwin, which comes with the KDE desktop environment. Years ago I tried to test GNUstep code with a lot of different window managers but I gave up on that long ago.

Comment (trunkmaster) [reply Source/NSApplication.m]: I understand you. I guess you just don't see some appicon flickering (GSSuppressAppIcon?). I have some questions on Cairo backend implementation. What is better the method for it: PM or mail-list?
